pkg/i2gw/ingress2gateway_test.go

```diff
@@ -150,3 +151,47 @@ func Test_constructIngressesFromCluster(t *testing.T) {
 		})
 	}
 }
+
+func Test_constructProviders(t *testing.T) {
+	ingress := ingress(443, "ingress1", "namespace1")
+	testCases := []struct {
+		name           string
+		runtimeObjs    []runtime.Object
+		providers      []string
+		expectingError bool
```

Comment:
Maybe add an extra field named `expectedProviders`? Then you can compare the results with it.

Also, I don't think you really need the `runtimeObjs` here

Reply:
Yes, you are right,  it's redundancy. I will remove it at next commit.